### PR TITLE
feat(SR): SCOORD3D point annotations support for volume and stack viewports

### PR DESCRIPTION
**Pull request by igoroctaviano**
### Context
- https://github.com/OHIF/Viewers/pull/4315

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: packages/core/src/RenderingEngine/BaseVolumeViewport.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import vtkPiecewiseFunction from '@kitware/vtk.js/Common/DataModel/PiecewiseFunc
 
 import { vec2, vec3 } from 'gl-matrix';
 
+import * as metaData from '../metaData';
```
Comment (sedghi) [inline packages/core/src/RenderingEngine/BaseVolumeViewport.ts:8]: seems like this file does not have any meaninfgul change, please revert

**Comment by igoroctaviano** [comment]
@sedghi resolved

File: packages/core/src/RenderingEngine/StackViewport.ts [hunk 2]
```diff
@@ -2913,16 +2919,38 @@ class StackViewport extends Viewport implements IStackViewport, IImagesLoader {
...
       imageURI = imageId.substring(colonIndex + 1);
     }
-    return referencedImageId?.endsWith(imageURI);
+
+    const endsWith = referencedImageId?.endsWith(imageURI);
+    if (endsWith) {
+      return endsWith;
+    }
+
+    if (cameraFocalPoint && FrameOfReferenceUID) {
```
Comment (sedghi) [inline packages/core/src/RenderingEngine/StackViewport.ts:2939]: There is no scenario where we don't have cameraFocalPoint

File: packages/core/src/types/IViewport.ts [hunk 3]
```diff
@@ -69,6 +69,8 @@ export type ReferenceCompatibleOptions = {
    * not need to be provided.
    */
   imageURI?: string;
+  isFiltering?: boolean;
+  currentImageId?: string;
```
Comment (sedghi) [inline packages/core/src/types/IViewport.ts:73]: I don't think these belongs here

**Comment by IbrahimCSAE** [comment]
rebase from dev to fix the CI

**Comment by sedghi** [comment]
closing in favor of https://github.com/cornerstonejs/cornerstone3D/pull/1487

**Pull request closed without merging**